app/models/cinstance.rb

```diff
@@ -409,6 +411,10 @@ def change_plan_internal(new_plan)
     end
   end
 
+  def same_service
+    errors.add(:plan_id, :not_allowed) if plan&.service != service
```

Comment:
AccountPlan does not have service.

Reply:
That's only for applications.

Reply:
I mean this will be always true for AccountPlan:
`plan&.service != service`

Reply:
My bad, you are right, the account_contract will not have any service either.

Reply:
Also, this is in `Cinstance`. It does not apply to all types of contracts.